internal/uvm/clone.go

```diff
@@ -62,15 +70,22 @@ type UVMTemplateConfig struct {
 	UVMID string
 	// Array of all resources that will be required while making a clone from this template
 	Resources []Cloneable
+	// The OptionsWCOW used for template uvm creation
+	CreateOpts OptionsWCOW
 }
 
 // Captures all the information that is necessary to properly save this UVM as a template
 // and create clones from this template later. The struct returned by this method must be
 // later on made available while creating a clone from this template.
-func (uvm *UtilityVM) GenerateTemplateConfig() *UVMTemplateConfig {
+func (uvm *UtilityVM) GenerateTemplateConfig() (*UVMTemplateConfig, error) {
+	if _, ok := uvm.createOpts.(OptionsWCOW); !ok {
+		return nil, fmt.Errorf("template config can only be created for a WCOW uvm")
+	}
+
 	// Add all the SCSI Mounts and VSMB shares into the list of clones
 	templateConfig := &UVMTemplateConfig{
```

Comment:
We should probably version the overall `UVMTemplateConfig` as well, so we can change the format in the future.

Reply:
I have made the changes as per your comments and I have added a new commit on this PR for ease of reviewing. I will squash that commit before merging.